src/FrameSemantics.cc

```diff
@@ -282,289 +282,511 @@ static Errors resolveModelPoseWithPlacementFrame(
-    // disconnected in this graph, the STATIC_MODEL was chosen as its frame
-    // type. A different frame type could potentially be used, but that adds
-    // more complexity to the validateFrameAttachedToGraph code.
-    _out = _out.AddScopeVertex(
-        "", "__root__", scopeContextName, sdf::FrameType::STATIC_MODEL);
-    rootId = _out.ScopeVertexId();
   }
 
-  const auto modelId = _out.AddVertex(_model->Name(), frameType).Id();
+  /// \brief Constructor that takes an sdf::InterfaceFrame
+  explicit FrameWrapper(const sdf::InterfaceFrame &_ifaceFrame)
+      : WrapperBase{_ifaceFrame.Name(), "Interface Frame", FrameType::FRAME},
+        rawPose(_ifaceFrame.PoseInAttachedToFrame()),
+        relativeTo(_ifaceFrame.AttachedTo()),
+        attachedTo(_ifaceFrame.AttachedTo()),
+        finalRelativeTo(attachedTo)
 
-  auto outModel = _out.AddScopeVertex(
-      _model->Name(), scopeContextName, scopeContextName, frameType);
-  const auto modelFrameId = outModel.ScopeVertexId();
+  {
+  }
 
-  // Add an aliasing edge from the model vertex to the
-  // corresponding vertex in the FrameAttachedTo graph of the child model,
-  auto &edge = outModel.AddEdge({modelId, modelFrameId}, true);
-  // Set the edge weight to 0 to indicate that this is an aliasing edge.
-  edge.SetWeight(0);
+  /// \brief Raw pose of the entity.
+  const ignition::math::Pose3d rawPose;
+  /// \brief The //pose/@relative_to attribute.
+  const std::string relativeTo;
+  /// \brief The //frame/@attached_to attribute.
+  const std::string attachedTo;
+  /// \brief The final @relative_to attribute. For sdf::Frame, this is set to
+  /// the attachedTo if the relativeTo is empty. For sdf::InterfaceFrame, it's
+  /// always set to attachedTo.
+  const std::string finalRelativeTo;
+};
 
-  // add link vertices
-  for (uint64_t l = 0; l < _model->LinkCount(); ++l)
+/// \brief Wrapper for sdf::Joint and sdf::InterfaceJoint
+struct JointWrapper: public WrapperBase
+{
+  /// \brief Constructor that takes an sdf::Joint
+  explicit JointWrapper(const sdf::Joint &_joint)
+      : WrapperBase{_joint.Name(), "Joint", FrameType::JOINT},
+        rawPose(_joint.RawPose()),
+        relativeTo(_joint.PoseRelativeTo()),
+        childName(_joint.ChildLinkName()),
+        finalRelativeTo(relativeTo.empty() ? childName : relativeTo)
+  {
+  }
+
+  /// \brief Constructor that takes an sdf::InterfaceJoint
+  explicit JointWrapper(const sdf::InterfaceJoint &_ifaceJoint)
+      : WrapperBase{_ifaceJoint.Name(), "Interface Joint", FrameType::JOINT},
+        rawPose(_ifaceJoint.PoseInChildFrame()),
+        relativeTo(_ifaceJoint.ChildName()),
+        childName(_ifaceJoint.ChildName()),
+        finalRelativeTo(childName)
+  {
+  }
+
+  /// \brief Raw pose of the entity.
+  const ignition::math::Pose3d rawPose;
+  /// \brief The //pose/@relative_to attribute.
+  const std::string relativeTo;
+  /// \brief The name of the child frame.
```

Comment:
This is the content of `//joint/child` so it would be child frame. I added that to the docs inhttps://github.com/ignitionrobotics/sdformat/pull/764/commits/59050f96134d6bf0cea07e35ab7916b1da7b70d9.